### PR TITLE
OpenAFS 1.8 fixups

### PR DESCRIPTION
**Pull request by spacefrogg**
###### Motivation for this change

Volume salvaging was partly broken. Parallel building is also broken.

###### Things done
Some fixes and improvements:
- FIX :: disable parallel building, as it is broken (in rare cases)
- FIX :: merge the `server` output with `out`, as it was breaking volume salvaging, due to hard-coded cross dependencies that got broken during fixup phase
- IMPROVEMENT :: add option to produce developer documentation

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by roberth** [comment]
Please update the commit author with `git config`, `git rebase -i`, its `e` command and `git commit --reset-author`, or update your github e-mail settings to include a match with the existing commits at http://github.com/settings/emails
Exceptions might be possible, but github behaves better when it matches.

**Comment by Mic92** [comment]
Could you incorporate https://github.com/NixOS/nixpkgs/pull/103860 ?

File: pkgs/servers/openafs/1.8/default.nix [hunk 1]
```diff
@@ -44,19 +47,23 @@ stdenv.mkDerivation {
 
     ./regen.sh
 
+    for i in $(grep -l -R '/usr/bin/file\b' .); do
+      echo "Patch /usr/bin/file in $i"
+      substituteInPlace $i --replace "/usr/bin/file" "${stdenv.lib.getBin file}/bin/file"
+    done
```
Comment (Mic92) [inline pkgs/servers/openafs/1.8/default.nix:53]: We tend to ignore this error since it happens in all autotools setups and does not harm.

Comment (Mic92) [inline pkgs/servers/openafs/1.8/default.nix:53]:
```suggestion
```

**Comment by spacefrogg** [comment]
Sure.

**Pull request merged**